ext/nokogiri/xml_namespace.c

```diff
@@ -34,21 +62,37 @@ static VALUE href(VALUE self)
   return NOKOGIRI_STR_NEW2(ns->href);
 }
 
+static int part_of_an_xpath_node_set_eh(xmlNsPtr node)
+{
+  return (node->next && ! Nokogiri_namespace_eh(node->next));
```

Comment:
Is this check really correct? I did not look into the libxml sources, but it looks strange to me.